staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/connection.go

```diff
@@ -123,7 +123,10 @@ func (c *connection) Close() error {
 func (c *connection) RemoveStreams(streams ...httpstream.Stream) {
 	c.streamLock.Lock()
 	for _, stream := range streams {
-		delete(c.streams, stream.Identifier())
+		// It may be possible that the provided stream is nil if timed out.
+		if stream != nil {
+			delete(c.streams, stream.Identifier())
+		}
```

Comment:
I feel that adding a judgment on the caller is also a good choice

https://github.com/kubernetes/kubernetes/blob/eee6e54ecf364ee0529788eb5ad72d4ffaccf1b4/pkg/kubelet/cri/streaming/portforward/httpstream.go#L168

Reply:
Hey @wzshiming, thank you for the review! Do you mean that we should move the nil check over to the `RemoveStreams()` invocation in favor of checking here?

Reply:
it's fine to check in both places, but I want this check to stay here, since it prevents all mistakes.

Reply:
Question: will the timed out stream not remain hanging around in the c.streams map, since the identity is no longer known when calling RemoveStreams?

I guess it gets cleaned up during Close, but since the unit test below does an explicit len(c.streams) check, just wondering if something else might use a similiar check to determine whether the connection can be closed?

Reply:
Yeah that's an excellent question, good reason to do reverts rather than another round of cherry-picks.

Reply:
Tested it locally with a modified version of CRI-O (using this vendored code) and it looks like that the connections are getting cleaned up if the timeout got reached.